File: src/tasks/cli/command/complete/CompleteParser.java
package tasks.cli.command.complete;

import omnia.data.cache.Memoized;
import omnia.data.structure.List;
import tasks.cli.parser.ParserUtil;
import tasks.cli.arguments.SimpleArguments;
import tasks.cli.parser.Parser;
import tasks.model.Task;

/** Command line argument parser for the Complete command. */
public final class CompleteParser extends SimpleArguments.Parser<CompleteArguments> {
  public CompleteParser(
      Memoized<Parser<? extends List<ParserUtil.ParseResult<Task>>>> taskParser) {
    super(taskParser, CompleteArguments::new);
  }
}

File: cli/src/me/strangepan/tasks/cli/feature/complete/CompleteParser.kt
package me.strangepan.tasks.cli.feature.complete

import java.util.function.Function
import omnia.data.cache.Memoized
import omnia.data.structure.List
import me.strangepan.tasks.cli.command.common.simple.SimpleParser
import me.strangepan.tasks.cli.parser.ParseResult
import me.strangepan.tasks.cli.parser.Parser
import me.strangepan.tasks.engine.model.Task

/** Command line argument parser for the Complete command.  */
class CompleteParser(
    taskParser: Memoized<out Parser<out List<out ParseResult<out Task>>>>) : SimpleParser<CompleteArguments>(Function { tasks: List<Task> -> CompleteArguments(tasks) }, taskParser)